runners/pm2.js

```diff
@@ -72,7 +73,15 @@ module.exports = {
 								if (exitCode == 0) {
 									next();
 								} else {
-									next(`Non-zero exit code: ${exitCode}`);
+									// We may be able to retrieve actual output from logs
+									var logPath = _.get(proc, '0.pm2_env.pm_out_log_path', '');
+									if (logPath) {
+										readLastLines.read(logPath, 1)
+											.then((lines) => next(lines))
```

Comment:
More helpful error logs for agents which fail with unexpected, unhandled promise rejections.